### PR TITLE
test: add e2e test scenarios

### PR DESCRIPTION
**Pull request by tomwwinter**
Closes https://github.com/Aam-Digital/aam-services/issues/8
Closes https://github.com/Aam-Digital/aam-services/issues/6

### CODE REVIEW

**Review by sleidig — APPROVED** [review]
Very nice structure, this looks easy to understand and easy to add more test scenarios :+1: 

Is there a code-climate [plugin for test coverage](https://docs.codeclimate.com/docs/configuring-test-coverage) for this, similar to what we have in ndb-core? Would be neat to see it as github PR checks (and track it in code-climate) instead of the multiple bot comments :blush: 

A ton of e2e test scenarios are still missing of course. Shall we create a separate issue for these?
(creating calculations, getting actual data calculated by sqs and all the webhook parts are critically to get tested ASAP)

File: application/aam-backend-service/src/test/kotlin/com/aamdigital/aambackendservice/container/TestContainers.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    Pair("COUCHDB_USER", "admin"),
+                    Pair("COUCHDB_PASSWORD", "docker"),
+                    Pair("COUCHDB_SECRET", "docker"),
+                )
+            )
+            .withExposedPorts(5984)
+
+//    @Container
+//    @JvmStatic
+//    val CONTAINER_SQS: GenericContainer<*> =
```
[the remark this answers is not in the document]
Comment (tomwwinter) [reply application/aam-backend-service/src/test/kotlin/com/aamdigital/aambackendservice/container/TestContainers.kt]: removed unused code

File: application/aam-backend-service/src/test/resources/cucumber/features/reporting/reportcalculation/start-report-calculation.feature [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Then the client waits for 15000 milliseconds
+        When the client calls GET /v1/reporting/report-calculation/ReportCalculation:2
+        Then the client receives an json object
+        Then the client receives status code of 200
+        Then the client receives value FINISHED_ERROR for property status
+
+    Scenario: Pending ReportCalculation is processed within 30 seconds
+        Given document ReportConfig:1 is stored in database app
+        Given document Config:CONFIG_ENTITY is stored in database app
+        Given _design document sqlite:config is stored in database app
```
Comment (sleidig) [inline application/aam-backend-service/src/test/resources/cucumber/features/reporting/reportcalculation/start-report-calculation.feature:46]: maybe not necessary?

**Review by sleidig — APPROVED** [review]

File: application/aam-backend-service/src/test/resources/cucumber/features/reporting/reportcalculation/start-report-calculation.feature (hunk 2 above)
Comment (tomwwinter) [reply application/aam-backend-service/src/test/resources/cucumber/features/reporting/reportcalculation/start-report-calculation.feature]: correct, should not be necessary. Code adapted and handed that case, removed design document creation in test.

**Pull request merged**